crypto/secp256k1/curve.go

```diff
@@ -116,6 +116,10 @@ func (BitCurve *BitCurve) IsOnCurve(x, y *big.Int) bool {
 // affineFromJacobian reverses the Jacobian transform. See the comment at the
 // top of the file.
 func (BitCurve *BitCurve) affineFromJacobian(x, y, z *big.Int) (xOut, yOut *big.Int) {
+	if z.Sign() == 0 {
```

Comment:
The purpose of adding `if z.Sign() == 0` here is to prevent that `z` and `BitCurve.P` are not relatively prime which would lead to `zinv == nil` , right? If so, besides z.Sign() == 0 , there are other conditions/situations which may cause `z` and `BitCurve.P` to be not relatively prime. In other words, `z.Sign() == 0` is not the only reason to cause `z` and `BitCurve.P` to be not relatively prime.

Maybe you can furthur check source code of GCD method of big.Int as following:

```
// GCD sets z to the greatest common divisor of a and b, which both must
// be > 0, and returns z.
// If x or y are not nil, GCD sets their value such that z = a*x + b*y.
// If either a or b is <= 0, GCD sets z = x = y = 0.
func (z *Int) GCD(x, y, a, b *Int) *Int {
	if a.Sign() <= 0 || b.Sign() <= 0 {
		z.SetInt64(0)
		if x != nil {
			x.SetInt64(0)
		}
		if y != nil {
			y.SetInt64(0)
		}
		return z
	}

	return z.lehmerGCD(x, y, a, b)
}
```

Here `z` is passed in as `a` , and `BitCurve.P` is passed in as `b`. As long as `a` and `b` are not relatively prime, the return value would not be equal to 1.

Reply:
Actually, I added two curve points which have the same x value, it Invoked `Add` function and `Add` function invoked `affineFromJacobian`, the result is infinity and should return (0,0), but the `affineFromJacobian` function panic. I can deal the situation in my own code, but should there something be done in the curve code？

Reply:
Maybe you could provide more detailed log about how panic happened in order to find out the root cause of the panic.